flink-connector/src/main/java/com/datastrato/gravitino/flink/connector/hive/HivePropertiesConverter.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+
+package com.datastrato.gravitino.flink.connector.hive;
+
+import static com.datastrato.gravitino.catalog.hive.HiveCatalogPropertiesMeta.METASTORE_URIS;
+
+import com.datastrato.gravitino.flink.connector.PropertiesConverter;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Maps;
+import java.util.Map;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.table.catalog.CommonCatalogOptions;
+import org.apache.flink.table.catalog.hive.HiveCatalog;
+import org.apache.flink.table.catalog.hive.factories.HiveCatalogFactoryOptions;
+import org.apache.hadoop.hive.conf.HiveConf;
+
+public class HivePropertiesConverter implements PropertiesConverter {
+
+  private HivePropertiesConverter() {}
+
+  public static final HivePropertiesConverter INSTANCE = new HivePropertiesConverter();
+
+  @Override
+  public Map<String, String> toGravitinoCatalogProperties(Configuration flinkConf) {
+    Map<String, String> gravitinoProperties = Maps.newHashMap();
+
+    // set metastore.uris
+    String hiveConfDir = flinkConf.get(HiveCatalogFactoryOptions.HIVE_CONF_DIR);
+    HiveConf hiveConf = HiveCatalog.createHiveConf(hiveConfDir, null);
+    String metastoreUris =
+        Preconditions.checkNotNull(
+            hiveConf.get(HiveConf.ConfVars.METASTOREURIS.varname),
+            "%s should not be null",
+            HiveConf.ConfVars.METASTOREURIS.varname);
+    gravitinoProperties.put(METASTORE_URIS, metastoreUris);
+
+    for (Map.Entry<String, String> entry : flinkConf.toMap().entrySet()) {
```

Comment:
what if there are duplicate key with different prefix? like `key`,  `flink.bypass.key`

Reply:
The configuration from the Gravitino has higher priority than that from the `hive-conf-dir`. And the key without `flink.bypass` from the Gravitino will be ignored. So the `hive.metastore.uris` must be saved in `hive-conf-dir`, and it will be converter to the `metastore.uri` for creating the Gravitino Catalog. we can also set `flink.bypass.hive.metastore.uris` to overwrite the `hive.metastore.uris` saved in `hive-conf-dir`.